### DISCUSSION TITLE
How can I refresh without breaking?

### DISCUSSION THREAD

**Discussion by gbizzotto** [hneemann/Digital#584] [Q&A] [answered]
I'm finishing up a brainfuck CPU that'll output to a custom LED display.
I'd like to see the display evolve with each character output, instead of having only a breakpoint after everything is done.
It would also be useful for seeing some progress (in my case, LEDs getting lit up), since the whole thing takes forever to run. Breakpoints are useful, but they force the user to click run again each time.

Is there a built-in component to make Digital refresh the display (lit wires, memory windows, 7-segments, LEDs, everything as when running step-by-step), without stopping/breaking the simulation?

**Comment by hneemann** [comment]
Hmm! There is already an automatic refresh. See the ProcessorVonNeumann example.
There must be some kind of misunderstanding.
The LED display is intended for operation with a logic for continuous repetition of the output, as it is also necessary with a simple real LED matrix. But the display follows a non High-Z input immediately.

**Comment by gbizzotto** [comment]
I've looked around and couldn't figure out what made ProcessorVonNeumann refresh the circuit while running. Is it the mere presence of a built-in LED display?
So far my circuits only get refreshed when paused, on a breakpoint or running step by step.
As for the LED display, i haven't used the built-in one. I actually made my own out of individual RGB LEDs and custom registers to hold each individual LED's color after it has been set.
And it's not updating on the fly.

Some explanation: I hadn't tried the built-in LED display. I wanted to build everything from scratch, including memory, registers, counters and everyting. I ended up using the built-in ones because of performance issues. My ram module (64kiB,  2.5 million gates) did not run (Digital silently gives up during the initialization of the simulation it seems). The 4k one was slow enough.

**Reply by hneemann** [reply]
I have tested the RGB LED again and it seems to work fine. Could you send me a minimal example of the problem? Then I could try to find out what is the cause.

> My ram module (64kiB, 2.5 million gates) did not run (Digital silently gives up during the initialization of the simulation it seems).

Please take a look at this [issue](https://github.com/hneemann/Digital/issues/278#issuecomment-496887657).

**Comment by gbizzotto** [comment, marked as answer]
Ah, the difference is I hadn't checked "start real time clock".
Thanks for your help (:
